docs/standard/serialization/system-text-json/custom-contracts.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Custom serialization and deserialization contracts
+description: "Learn how to write your own contract resolution logic to customize the JSON contract for a type."
+ms.date: 09/26/2022
+---
+# Customize a JSON contract
+
+The <xref:System.Text.Json?displayProperty=fullName> library constructs a JSON *contract* for each .NET type, which defines how the type should be serialized and deserialized. The contract is derived from the type's shape, which includes characteristics such as its properties and fields and whether it implements the <xref:System.Collections.IEnumerable> or <xref:System.Collections.IDictionary> interface. Types are mapped to contracts either at run time using reflection or at compile time using the source generator.
+
+Starting in .NET 7, you can customize these JSON contracts to provide more control over how types are converted into JSON and vice versa. The following list shows just some examples of the types of customizations you can make to serialization and deserialization:
+
+- Serialize private fields and properties.
+- Support multiple names for a single property (for example, if a previous library version used a different name).
+- Ignore properties with a specific name, type, or value.
+- Distinguish between explicit `null` values vs. lack of a value in the JSON payload.
```

Comment:
It might be worth including an example adding support for third-party/custom attributes, as I anticipate this will be a powerful use case for the feature.

Reply:
Ah, I just saw this has been incorporated in the private fields example.